### ISSUE TITLE
Make top speeds on a time limit

### ISSUE THREAD

**Issue by gamerdude23** [ChainedDev/CityLife#592]
**Is your feature request related to a problem? Please describe.**
A lot of people are just pushing their car to the max, and it isn't very realistic. 

**Describe the feature you'd like**
Add an engine oil temp gauge or something similar, so if you are pushing say 180 in your car, you can't keep that speed forever, it would last a while like 10 mins
When you upgrade the engine in your car, the time that your car can last at it's top speed increases. 
Increase top speed as kind of a balance for it. 
Say your car's top speed is 170, if you keep holding that for 10 mins, and the engine tempt gauge hits it's max, your engine has a high chance of blowing. It isn't 100% chance that your car engine will blow, but chances are high, and pushing it for longer would increase the chances of your engine blowing. 
tl:dr
Improve sports/hyper car top speed, but make it so that running the engine at it's peak for a certain time, it will blow, and after every upgrade, the time gets lengthened. Ex. Stock: 6 mins, maxed engine, 16 mins.
**Describe alternatives you've considered**
None

**Additional context**
Probably mainly add to sports cars, cheaper cars like panto can stay the same and not overheat. 
Would add more reason for engine work done, as car upgrades don't really improve cars that much. 
Maybe would encourage drivers to not be going full speed all the time in their car.

**Comment by Dollar82** [comment]
I don't know anywhere you can hold that speed for that long 😂

**Comment by gamerdude23** [comment]
With a built engine it can be possible, a car from the 80s is known to keep it's top speed of 200 mph+ for 15 mins+ at a time.

**Comment by Dollar82** [comment]
I dont know anywhere in San Andreas you can hold speed that long😂
